src/libraries/System.Net.Sockets/src/System/Net/Sockets/DynamicWinsockMethods.cs

```diff
@@ -192,8 +192,9 @@ internal unsafe bool ConnectEx(SafeSocketHandle socketHandle, IntPtr socketAddre
                     socketHandle.DangerousAddRef(ref socketHandle__addRefd);
                     __socketHandle_gen_native = socketHandle.DangerousGetHandle();
                     fixed (int* __bytesSent_gen_native = &bytesSent)
+                    fixed (void* socketAddressPtr = &MemoryMarshal.GetReference(socketAddress))
                     {
-                        __retVal_gen_native = ((delegate* unmanaged<IntPtr, IntPtr, int, IntPtr, int, int*, NativeOverlapped*, int>)_target)(__socketHandle_gen_native, socketAddress, socketAddressSize, buffer, dataLength, __bytesSent_gen_native, overlapped);
+                        __retVal_gen_native = ((delegate* unmanaged<IntPtr, void*, int, IntPtr, int, int*, NativeOverlapped*, int>)_target)(__socketHandle_gen_native, socketAddressPtr, socketAddress.Length, buffer, dataLength, __bytesSent_gen_native, overlapped);
```

Comment:
So `ConnectEx` will do a copy of the buffer internally, and not hold into the pointer for the time of the async call?

Reply:
[Yes.](https://github.com/dotnet/runtime/issues/86513#:~:text=needs%20to%20be%20preserved%20only%20through%20initial%20call%20and%20not%20through%20life%20of%20the%20overlapped%20IO.%20(confirmed%20with%20Windows%20networking%20team))